src/main/java/org/opensearch/knn/index/mapper/LuceneFieldMapper.java

```diff
@@ -61,45 +66,57 @@ public class LuceneFieldMapper extends KNNVectorFieldMapper {
             );
         }
 
-        this.fieldType = KnnVectorField.createFieldType(dimension, vectorSimilarityFunction);
+        this.fieldType = vectorDataType.createKnnVectorFieldType(dimension, vectorSimilarityFunction);
 
         if (this.hasDocValues) {
-            this.vectorFieldType = buildDocValuesFieldType(this.knnMethod.getKnnEngine());
+            this.vectorFieldType = vectorDataType.buildDocValuesFieldType(this.knnMethod.getKnnEngine());
         } else {
             this.vectorFieldType = null;
         }
     }
 
-    private static FieldType buildDocValuesFieldType(KNNEngine knnEngine) {
-        FieldType field = new FieldType();
-        field.putAttribute(KNN_ENGINE, knnEngine.getName());
-        field.setDocValuesType(DocValuesType.BINARY);
-        field.freeze();
-        return field;
-    }
-
     @Override
     protected void parseCreateField(ParseContext context, int dimension) throws IOException {
 
         validateIfKNNPluginEnabled();
         validateIfCircuitBreakerIsNotTriggered();
 
-        Optional<float[]> arrayOptional = getFloatsFromContext(context, dimension);
-
-        if (arrayOptional.isEmpty()) {
-            return;
-        }
-        final float[] array = arrayOptional.get();
-
-        KnnVectorField point = new KnnVectorField(name(), array, fieldType);
-
-        context.doc().add(point);
-        if (fieldType.stored()) {
-            context.doc().add(new StoredField(name(), point.toString()));
-        }
-
-        if (hasDocValues && vectorFieldType != null) {
-            context.doc().add(new VectorField(name(), array, vectorFieldType));
+        if (VectorDataType.BYTE.equals(vectorDataType)) {
+            Optional<byte[]> bytesArrayOptional = getBytesFromContext(context, dimension);
+            if (bytesArrayOptional.isEmpty()) {
+                return;
+            }
+            final byte[] array = bytesArrayOptional.get();
+            KnnByteVectorField point = new KnnByteVectorField(name(), array, fieldType);
+            context.doc().add(point);
+            if (fieldType.stored()) {
+                context.doc().add(new StoredField(name(), point.toString()));
+            }
+            if (hasDocValues && vectorFieldType != null) {
+                context.doc().add(new VectorField(name(), array, vectorFieldType));
+            }
```

Comment:
once the vector field is create this common part of the code is common and can be shared. So, lets do that and remove this duplication

Reply:
refactored for storedFields. But, for docValues we can't do it because the datatype of the array is different for both.